src/im/keyboard/keyboard.cpp

```diff
@@ -327,6 +327,16 @@ std::vector<InputMethodEntry> KeyboardEngine::listInputMethods() {
     return result;
 }
 
+void KeyboardEngine::activate(const InputMethodEntry &entry,
```

Comment:
I think this should be done in the state constructor.

Engine::activate is basically focus in.

Reply:
Thank you for the advice.
I have fixed this.